master/pkg/searcher/search_method.go

```diff
@@ -20,33 +22,35 @@ type SearchMethod interface {
 	initialOperations(ctx context) ([]Operation, error)
 	// trialCreated informs the searcher that a trial has been created as a result of a Create
 	// operation.
-	trialCreated(ctx context, requestID RequestID) ([]Operation, error)
+	trialCreated(ctx context, requestID model.RequestID) ([]Operation, error)
 	// trainCompleted informs the searcher that the training workload initiated by the same searcher
 	// has completed. It returns any new operations as a result of this workload completing.
-	trainCompleted(ctx context, requestID RequestID, train Train) ([]Operation, error)
+	trainCompleted(ctx context, requestID model.RequestID, train Train) ([]Operation, error)
 	// checkpointCompleted informs the searcher that the checkpoint workload initiated by the same
 	// searcher has completed. It returns any new operations as a result of this workload
 	// completing.
 	checkpointCompleted(
-		ctx context, requestID RequestID, checkpoint Checkpoint, metrics workload.CheckpointMetrics,
+		ctx context, requestID model.RequestID, checkpoint Checkpoint, metrics workload.CheckpointMetrics,
 	) ([]Operation, error)
 	// validationCompleted informs the searcher that the validation workload initiated by the same
 	// searcher has completed. It returns any new operations as a result of this workload
 	// completing.
 	validationCompleted(
-		ctx context, requestID RequestID, validate Validate, metrics workload.ValidationMetrics,
+		ctx context, requestID model.RequestID, validate Validate, metrics workload.ValidationMetrics,
 	) ([]Operation, error)
 	// trialClosed informs the searcher that the trial has been closed as a result of a Close
 	// operation.
-	trialClosed(ctx context, requestID RequestID) ([]Operation, error)
+	trialClosed(ctx context, requestID model.RequestID) ([]Operation, error)
 	// progress returns experiment progress as a float between 0.0 and 1.0. As search methods
 	// receive completed workloads, they should internally track progress.
 	progress(totalUnitsCompleted float64) float64
 	// trialExitedEarly informs the searcher that the trial has exited earlier than expected.
 	trialExitedEarly(
-		ctx context, requestID RequestID, exitedReason workload.ExitedReason,
+		ctx context, requestID model.RequestID, exitedReason workload.ExitedReason,
 	) ([]Operation, error)
-	// SearchMethod embeds the InUnits interface because it is in terms of a specific unit.
+	// NOTE: searcher has to handle deser because only it knows appropriate types.
```

Comment:
deserializing. this was from way before. the comment basically is searcher has handle serializing/deserializing itself since the experiment is unaware of the type of searcher it is.